cpp/src/parquet/encryption/crypto_factory.h

```diff
@@ -148,7 +148,7 @@ class PARQUET_EXPORT CryptoFactory {
       int dek_length, const std::string& column_keys, FileKeyWrapper* key_wrapper);
 
   /// Key utilities object for kms client initialization and cache control
-  KeyToolkit key_toolkit_;
+  std::shared_ptr<KeyToolkit> key_toolkit_;
```

Comment:
```suggestion
  std::shared_ptr<KeyToolkit> key_toolkit_ = std::make_shared<KeyToolkit>();
```
And remove the explicit ctor at line 103. WDYT?